Authorization/Authorization/Presentation/SocialSign/SocialSignViewModel.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  SocialSignViewModel.swift
+//  Authorization
+//
+//  Created by Eugene Yatsenko on 11.10.2023.
+//
+
+import SwiftUI
+import Core
+import AuthenticationServices
+import FacebookLogin
+import GoogleSignIn
+import MSAL
+import Swinject
+
+enum SocialResult {
+    case apple(AppleCredentials)
+    case facebook(LoginManagerLoginResult)
+    case google(GIDSignInResult)
+    case microsoft(MSALAccount, String)
+
+    var backend: String {
+        switch self {
+        case .apple:
+            return "apple-id"
+        case .facebook:
+            return "facebook"
+        case .google:
+            return "google-oauth2"
+        case .microsoft:
+            return "azuread-oauth2"
+        }
+    }
+}
+
+final public class SocialSignViewModel: ObservableObject {
+
+    // MARK: - Properties
+
+    private var completion: ((Result<SocialResult, Error>) -> Void)
+    private let config: ConfigProtocol
+
+    init(
+        config: ConfigProtocol,
+        completion: @escaping (Result<SocialResult, Error>) -> Void
+    ) {
+        self.config = config
+        self.completion = completion
+    }
+
+    private let appleSingInProvider: AppleSingInProvider  = .init()
+    private let googleSingInProvider: GoogleSingInProvider = .init()
+    private let facebookSingInProvider: FacebookSingInProvider = .init()
+    private let microsoftSingInProvider: MicrosoftSingInProvider = .init()
+
+    private var topViewController: UIViewController? {
+        UIApplication.topViewController()
+    }
+
+    // MARK: - Public Properties
+
+    var isFaceboolEnabled: Bool {
+        config.facebook.enabled
+    }
+
+    var isGoogleEnabled: Bool {
+        config.google.enabled
+    }
+
+    var isMicrosoftEnabled: Bool {
+        config.microsoft.enabled
+    }
+
+    var isAppleSignInEnabled: Bool {
```

Comment:
Although it's fine, but how about omitting `is` from names, making it like facebookEnabled? it will make it more simple/